deploy/sample-backup-configs/sample_config_build_recipes.yaml

```diff
@@ -0,0 +1,104 @@
+# Build-time design configuration for the FireSim Simulation Manager
+# See docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst for documentation of all of these params.
+
+# this file contains sections that describe hardware designs that /can/ be built.
+# edit config_build.yaml to actually "turn on" a config to be built when you run
+# buildafi
+
+###########
+# Schema:
+###########
+# <NAME>:
+#    DESIGN: <>
+#    TARGET_CONFIG: <>
+#    deploy-triplet: null
+#    PLATFORM_CONFIG: Config
+#    post-build-hook: null
+#    # REQUIRED: TODO: Will move to "run platform" work
+#    s3-bucket-name: firesim-AWSUSERNAME
+#    # OPTIONAL: Specific build platform to use. If unspecified, default to "default-build-farm"
+#    build-farm: default-build-farm
```

Comment:
I know you had a mini-conversation with @timsnyder-siv  about that but github is not letting me reply in there.

What I'd strongly prefer is a "default-build-farm" key in `config_build.yaml` that refers to a specific build farm definition in `config_build_farm.yml`, instread of choosing the build farm definition with this specific name.

Reply:
So to be clear, you want a required key in `_build.yaml` say `default-build-farm` that points to the name of a build farm in `_build_farm.yaml` (probably named something else other than "default-build-farm" now that that is a key). This key is then used for all builds? What happens if the user wants to use multiple different build farms at the same time? Is there an `override-build-farm` key in the build recipe that does this?

Reply:
To control the scope creep for now having a level of indirection is all i'm asking for: basically have `default-build-farm` be a reference to one of the other definitions and something the user can change (i.e., a key-value pair in  `config_build_farm.yml` called `default-build-farm:<build farm definition name>` at the top of the file). 

In my initial comment i was thinking choosing a default build farm should be a config-build thing, but we should leave that discussion to the next PR. 

----- 

I think this is revealing the limitation in the names we're using.

When you say use multiple build farms, do you mean use two definitions in config_build_farm.yaml to provide hosts for a single run of `firesim buildafi`? 

The way i see it one `firesim buildafi` invocation -> one build farm (instance). If one build farm (definition) in build-farm.yml is insufficient to do all the builds you want, you can compose definitions/providers/w.e, but you're still left with one build farm (instance).

Reply:
> but github is not letting me reply in there.

@davidbiancolin , I guess you can't reply to other conversations while you're in the middle of doing a review.  This seems like a bug in the UI but I'm sure someone thinks it is a feature.

Reply:
> When you say use multiple build farms, do you mean use two definitions in config_build_farm.yaml to provide hosts for a single run of `firesim buildafi`?

> The way i see it one `firesim buildafi` invocation -> one build farm (instance). If one build farm (definition) in build-farm.yml is insufficient to do all the builds you want, you can compose definitions/providers/w.e, but you're still left with one build farm (instance).

Yup. This is what I'm referring to (for ex. building on EC2 and locally at the same time). I see where the confusion around the naming is (one run of `buildafi` is associated with one `buildfarm` - how that build farm is composed should be named differently). To move this PR forward, I'll just enforce that only 1 build farm can be active for a `buildafi` run, and instance types cannot be mixed (i.e. you can't run builds on EC2 and with user-provided IP addresses).